pkg/netpol/eval/resources.go

```diff
@@ -207,52 +207,67 @@ func (pe *PolicyEngine) upsertNamespace(ns *corev1.Namespace) error {
 	return nil
 }
 
-// checkLegalOwnerRefNameToPodLabels returns error if there are pod resources with same ownerReferences name but different labels
-func (pe *PolicyEngine) checkLegalOwnerRefNameToPodLabels(newPod *k8s.Pod) error {
+// checkConsistentLabelsForPodsOfSameOwner returns error if there are pod resources with same ownerReferences name but different labels
+func (pe *PolicyEngine) checkConsistentLabelsForPodsOfSameOwner(newPod *k8s.Pod) error {
 	if newPod.Owner.Name == "" { // the new pod does not have owner references
 		return nil
 	}
-	if _, ok := pe.ownersToLabelsMap[newPod.Namespace]; !ok { // add the new namespace to the map
-		pe.ownersToLabelsMap[newPod.Namespace] = make(map[string]*k8s.Pod)
+	if _, ok := pe.podOwnersToRepresentativePodMap[newPod.Namespace]; !ok { // add the new namespace to the map
+		pe.podOwnersToRepresentativePodMap[newPod.Namespace] = make(map[string]*k8s.Pod)
 	}
-	firstPod, ok := pe.ownersToLabelsMap[newPod.Namespace][newPod.Owner.Name]
+	firstPod, ok := pe.podOwnersToRepresentativePodMap[newPod.Namespace][newPod.Owner.Name]
 	if !ok { // add the new ownerReference with this new pod
-		pe.ownersToLabelsMap[newPod.Namespace][newPod.Owner.Name] = newPod
+		pe.podOwnersToRepresentativePodMap[newPod.Namespace][newPod.Owner.Name] = newPod
 		return nil
 	}
 	// compare the owner first pod's labels with new pod's Labels
-	return diffBetweenPodsLabels(firstPod, newPod)
+	if key, firstVal, newVal := diffBetweenPodsLabels(firstPod, newPod); key != "" {
+		return writeLabelsDifferencesError(firstPod, newPod, key, firstVal, newVal) // err
+	}
+	return nil
 }
 
-// helper: given two pods of same owner, returns an error if there are diffs between the pods' labels maps
-func diffBetweenPodsLabels(firstPod, newPod *k8s.Pod) error {
+// helper: writeLabelsDifferencesError computes the error message of the pod's different labels
+func writeLabelsDifferencesError(firstPod, newPod *k8s.Pod, key, firstVal, newVal string) error {
```

Comment:
The function name is misleading, as it does not write anything. Consider changing to `generateLabelsDiffError`